Dockerfile

```diff
@@ -1,41 +1,43 @@
-ARG ALPINE_VERSION=3.10
-ARG GO_VERSION=1.13.4
-ARG GRPC_GATEWAY_VERSION=1.12.2
-ARG GRPC_JAVA_VERSION=1.26.0
-ARG GRPC_VERSION=1.26.0
-ARG PROTOC_GEN_GO_VERSION=1.3.2
-ARG PROTOC_GEN_GOGO_VERSION=ba06b47c162d49f2af050fb4c75bcbc86a159d5c
+ARG ALPINE_VERSION=3.11
+ARG GO_VERSION=1.14.2
+ARG GRPC_GATEWAY_VERSION=1.14.3
+ARG GRPC_JAVA_VERSION=1.28.1
+ARG GRPC_VERSION=1.28.1
```

Comment:
what does this mean? Is this grpc-go version?

Reply:
Yes, since only Java and Go are in separate repos, we could keep those on old versions as long as the interface to protoc is compatible. All other (cpp, csharp, python) are generated from the regular grpc/protobuf repo and bound to the version that also creates protoc as there are lots of build dependencies.

We could try to find out if it’s possible to create the static libs for protoc and use different versions of the protoc-gen-x to build. Don’t know the compatibility as especially for the c based languages, they are intended to be build at one time.

Reply:
I spent half a day doing tests today. Since we are compiling `libproto` and `libprotobuf` from this base, all generators build on top of that use this. The regular `protoc` generators (cpp, csharp, java, js, objc, php, python, ruby) are already compiled into it and not distributed as plugins. For Java, only the gRPC part is a separate repository. All others are in one repository.

Technically, I can keep Java gRPC on the old version, but this might break since the protos for Java will be generated against a newer version. Only the go generators (proto + grpc) are completely separated and could be kept completely on the old version if preferred.

Right now, I'm looking into having a second build of protoc contained in the image and having the `protoc-wrapper` making the differentiation. A bit ugly, but seems like the only option if we do not want to have one protoc for all generators (like we do right now).